Actions/AL-Go-Helper.ps1

```diff
@@ -20,7 +20,7 @@ $defaultCICDPullRequestBranches = @( 'main' )
 $runningLocal = $local.IsPresent
 $defaultBcContainerHelperVersion = "preview" # Must be double quotes. Will be replaced by BcContainerHelperVersion if necessary in the deploy step - ex. "https://github.com/organization/navcontainerhelper/archive/refs/heads/branch.zip"
 $microsoftTelemetryConnectionString = "InstrumentationKey=84bd9223-67d4-4378-8590-9e4a46023be2;IngestionEndpoint=https://westeurope-1.in.applicationinsights.azure.com/"
-$notSecretProperties = @("Scopes","TenantId","BlobName","ContainerName","StorageAccountName","ServerUrl","ppUserName")
+$notSecretProperties = @("Scopes","TenantId","BlobName","ContainerName","StorageAccountName","ServerUrl","ppUserName","ErrorAction")
```

Comment:
ErrorAction is still used in AZURE_CREDENTIALS and this makes sense.
We can build without secrets in a pull request, meaning that if we build from a branch, which doesn't have access to the keyvault - then we will still build - just ignore that we cannot reach secrets from the keyvault.

Reply:
> just ignore that we cannot reach secrets from the keyvault.

But presumably there are secrets in the key vault that are needed for something. Otherwise, you wouldn't set up the key vault connection 🤔 Why would you want to silently continue without getting the secrets? And even if we do, there is no guarantee the workflow wouldn't just fail later on

Reply:
Today secrets are mainly used for signing, deploying, delivering and resolving dependencies.

This is mainly a problem when using federated credentials, where you have to specify which branches can authenticate. 

Without this - I cannot create a branch in bcsamples-bingmaps.appsource to do feature work - even though that branch doesn't need any secrets at all.

With this - I can create a branch and do feature work - I cannot sign, deploy or deliver from that branch - which is fine.

If you have secrets for resolving dependencies - these could be added as github secrets and allow branches to build still.

Reply:
> Without this - I cannot create a branch in bcsamples-bingmaps.appsource to do feature work

You mean, you can't run CI/CD from a feature branch? Federated credentials support authenticating from pull requests so as long as that's set up you should be fine
![image](https://github.com/microsoft/AL-Go/assets/117829001/10f7dfe8-0c5e-4222-9e5d-c7f9f4b8dc8d)

Reply:
Feature branch or a release branch - not a pull request